pkg/spanstat/spanstat.go

```diff
@@ -37,10 +45,14 @@ func (s *SpanStat) Start() {
 func (s *SpanStat) End(success bool) {
 	if !s.spanStart.IsZero() {
 		d := time.Since(s.spanStart)
-		if success {
-			s.successDuration += d
+		if d >= time.Duration(0) {
+			if success {
+				s.successDuration += d
+			} else {
+				s.failureDuration += d
+			}
 		} else {
-			s.failureDuration += d
+			log.WithField("duration", d).Warn("Duration is negative on End")
```

Comment:
We need a stack trace for this to be useful

Reply:
Fixed.